### PR TITLE
Update docs for volume_driver option

### LINKED ISSUE #2960: volume_driver in service config is unnecessary in v2 format
I believe `volume_driver` is only relevant for named volumes, so the driver would need to be specified in the `volumes` section, not the config section.

I think we can remove `volume_driver` from the service config for v2.

### PR DESCRIPTION
**Pull request by shin-**
Fixes #2960

### CODE REVIEW

File: docs/compose-file.md [hunk 1]
```diff
@@ -552,7 +552,7 @@ limit as an integer or soft/hard limits as a mapping.
         soft: 20000
         hard: 40000
 
-### volumes, volume\_driver
+### volumes
 
 Mount paths or named volumes, optionally specifying a path on the host machine
 (`HOST:CONTAINER`), or an access mode (`HOST:CONTAINER:ro`). Named volumes can
```
Comment (dnephin) [inline docs/compose-file.md]: There's a line just below this that says

> top-level `volumes` key, but this is optional - the Docker Engine will create the volume if it doesn't exist.

I think we need to move that to a **Note**: In V1 ...

**Comment by dnephin** [comment]
Thanks for picking this up!

File: docs/compose-file.md (hunk 1 above)
Comment (shin-) [inline docs/compose-file.md]: Hm, yeah. There's also no top-level volume key in v1, so I think that part needs a small rewrite.

Comment (shin-) [inline docs/compose-file.md]: Here's what I have:

> Mount paths or named volumes, optionally specifying a path on the host machine
(`HOST:CONTAINER`), or an access mode (`HOST:CONTAINER:ro`).

> With the [version 2 file format](#version-2), named volumes must be specified
with the [top-level `volumes` key](#volume-configuration-reference). If you're
using [version 1](#version-1), named volumes will be created implicitly when
declared in this section.

WDYT?

Comment (dnephin) [inline docs/compose-file.md]: works for me

**Comment by dnephin** [comment]
It turns out I was wrong about this. @thaJeztah pointed me at https://github.com/docker/docker/pull/19190

So apparently we need to keep `volume_driver`, but we should document that it only applies to container volumes in v2, and not named volumes.

**Comment by shin-** [comment]
Updated :+1: Now only a docs change.

File: docs/compose-file.md [hunk 2]
```diff
@@ -580,11 +580,16 @@ should always begin with `.` or `..`.
...
-a `volume_driver`.
+If you do not use a host path, you may specify a `volume_driver`.
 
     volume_driver: mydriver
 
+Note that for [version 2 files](#version-2), this driver
+will not apply to named volumes (you should use the `driver` option when
+[declaring the volume](#volume-configuration-reference) instead).
+For [version 1](#version-1), both named volumes and container volumes will
+be afected.
```
Comment (aanand) [inline docs/compose-file.md]: Typo: "afected". But actually, I think better wording would be: "For version 1, both named volumes and container volumes will use the specified driver".

Comment (shin-) [inline docs/compose-file.md]: Fixed :+1:

**Comment by dnephin** [comment]
We lost the fix to the section above where it says "Named volumes can
 be specified with the
 [top-level `volumes` key](#volume-configuration-reference), but this is
 optional"

**Comment by shin-** [comment]
Good point, I updated it again.

**Comment by dnephin** [comment]
LGTM!

**Pull request merged**